backend/inspirehep/pidstore/cli.py

```diff
@@ -0,0 +1,140 @@
+    ).count()
+    if length_external and current_app.config["FEATURE_FLAG_ENABLE_BAI_CREATION"]:
+        click.echo(
+            "There are external BAIs detected in the Pidstore while requesting for creating missing BAIs."
+            " This is not allowed."
+            " Please register all existing BAIs first (by running this command without '--create' flag)."
+        )
+        return
+    start_not_external_bais_count = PersistentIdentifier.query.filter_by(
+        pid_type="bai", pid_provider="bai", status="R"
+    ).count()
+    click.echo(
+        f"{length} authors will be processed. {length_external} External BAI PIDs will be deleted."
+    )
+    if not yes_i_know:
+        click.confirm(
+            "Are you sure you want to continue?",
+            abort=True,
+        )
+    with click.progressbar(all_authors, length=length) as authors_bar:
+        for author_pid in authors_bar:
+            try:
+                removed_count = PersistentIdentifier.query.filter_by(
+                    object_uuid=author_pid.object_uuid,
+                    pid_type="bai",
+                    pid_provider="external",
+                ).delete()
+                LOGGER.debug(
+                    "Processing BAI for author",
+                    recid=author_pid.pid_value,
+                    removed_external_bais=removed_count,
+                )
+                author_record = AuthorsRecord.get_record(author_pid.object_uuid)
+                author_data = dict(author_record)
+                BAIMinter.mint(author_pid.object_uuid, author_data)
+                if author_data.get("ids") != author_record.get("ids") and create:
+                    author_record.update(author_data)
+                elif author_data.get("ids") != author_record.get("ids") and not create:
+                    db.session.rollback()
+                    click.echo(
+                        f"Unexpected change in author {author_pid.pid_value} metadata. Rolling back and exiting.",
+                        err=True,
+                    )
+                    return
+            except MissingSchema:
+                LOGGER.warning(
+                    "Record is missing Schema",
+                    recid=author_record.control_number,
+                    uuid=author_record.id,
+                )
+
+    new_not_external_bais_count = PersistentIdentifier.query.filter_by(
+        pid_type="bai", pid_provider="bai", status="R"
+    ).count()
+    new_external_bais_count = PersistentIdentifier.query.filter_by(
+        pid_type="bai", pid_provider="external", status="R"
+    ).count()
+
+    if not click.confirm(
+        f"There are {new_not_external_bais_count-start_not_external_bais_count} new bais with non external pid_provider added to the Pidstore."
+        f" {length_external-new_external_bais_count} BAIs with external pid_provider are deleted"
+        " Should those changes be commited to DB?"
+    ):
+        db.session.rollback()
+        return
```

Comment:
you're not testing for `yes_i_know` here, and you don't need to test for the output of `click.confirm`, as aborting the script will also rollback the transaction.

Reply:
I'm not testing for `yes_i_know` here as I want this to always display and ask for confirmation. This is really delicate operation and committing wrong data here will actually require a lot of work to undone it.
I am also aware that aborting will rollback, but here I want to be as much straightforward with the code as possible. This is dangerous operation here which at the end will actually modify Pidstore and records metadata in one big commit.